src/PackageFilter.php

```diff
@@ -40,15 +40,24 @@ public function __construct(IOInterface $io, string $symfonyRequire, Downloader
 
     /**
      * @param PackageInterface[] $data
+     * @param PackageInterface[] $lockedPackages
      *
      * @return PackageInterface[]
      */
-    public function removeLegacyPackages(array $data): array
+    public function removeLegacyPackages(array $data, array $lockedPackages): array
     {
         if (!$this->symfonyConstraints || !$data) {
             return $data;
         }
 
+        foreach ($lockedPackages as $i => $package) {
+            unset($lockedPackages[$i]);
+            $lockedPackages[$package->getName()] = [$package->getVersion()];
```

Comment:
to convert from a `list<Package>` to an indexed array `array<string, Package[]>`, I suggest creating a new array instead of reusing the same one. It will make the code much easier to understand

Reply:
updated